src/libcaf.h

```diff
@@ -89,7 +91,6 @@ typedef enum caf_deregister_t {
 caf_deregister_t;
 
 typedef void* caf_token_t;
-#ifdef GCC_GE_7
```

Comment:
Well, it would be safer to have this in #ifdef GCC_GE_8 (no typo here), then in GCC_GE_7, when the team-feature will only be available in GCC 8+. The other way around then is to make sure, that every use of team-features is cleanly wrapped in a similar ifdef.
I do not recommend to expose features of newer compilers, when we already guard against them at other locations.